src/test/java/io/cdap/plugin/couchbase/CouchbaseConfigTest.java

```diff
@@ -0,0 +1,581 @@
+      Assert.assertEquals("Reference name must be specified", validationFailure.getMessage());
+      Assert.assertEquals(1, validationFailure.getCauses().size());
+      ValidationFailure.Cause cause = validationFailure.getCauses().get(0);
+      Assert.assertEquals(Constants.Reference.REFERENCE_NAME, cause.getAttribute(CauseAttributes.STAGE_CONFIG));
+      Assert.assertEquals(MOCK_STAGE_NAME, cause.getAttribute(STAGE));
+    }
+  }
+
+  @Test
+  public void testValidateReferenceNameInvalid() {
+    try {
+      CouchbaseConfigBuilder.builder(VALID_CONFIG)
+        .setReferenceName("**********")
+        .build()
+        .validate(new MockFailureCollector(MOCK_STAGE_NAME));
+    } catch (ValidationException e) {
+      Assert.assertEquals(1, e.getFailures().size());
+      ValidationFailure validationFailure = e.getFailures().get(0);
+      Assert.assertEquals("Invalid reference name", validationFailure.getMessage());
+      Assert.assertEquals(1, validationFailure.getCauses().size());
+      ValidationFailure.Cause cause = validationFailure.getCauses().get(0);
+      Assert.assertEquals(Constants.Reference.REFERENCE_NAME, cause.getAttribute(CauseAttributes.STAGE_CONFIG));
+      Assert.assertEquals(MOCK_STAGE_NAME, cause.getAttribute(STAGE));
+    }
+  }
+
+  @Test
+  public void testValidateNodesNull() {
+    try {
+      CouchbaseConfigBuilder.builder(VALID_CONFIG)
+        .setNodes(null)
+        .build()
+        .validate(new MockFailureCollector(MOCK_STAGE_NAME));
+    } catch (ValidationException e) {
+      Assert.assertEquals(1, e.getFailures().size());
+      ValidationFailure validationFailure = e.getFailures().get(0);
+      Assert.assertEquals("Couchbase nodes must be specified", validationFailure.getMessage());
+      Assert.assertEquals(1, validationFailure.getCauses().size());
+      ValidationFailure.Cause cause = validationFailure.getCauses().get(0);
+      Assert.assertEquals(CouchbaseConstants.NODES, cause.getAttribute(CauseAttributes.STAGE_CONFIG));
+      Assert.assertEquals(MOCK_STAGE_NAME, cause.getAttribute(STAGE));
+    }
+  }
+
+  @Test
+  public void testValidateNodesEmpty() {
+    try {
+      CouchbaseConfigBuilder.builder(VALID_CONFIG)
+        .setNodes("")
+        .build()
+        .validate(new MockFailureCollector(MOCK_STAGE_NAME));
+    } catch (ValidationException e) {
+      Assert.assertEquals(1, e.getFailures().size());
+      ValidationFailure validationFailure = e.getFailures().get(0);
+      Assert.assertEquals("Couchbase nodes must be specified", validationFailure.getMessage());
+      Assert.assertEquals(1, validationFailure.getCauses().size());
+      ValidationFailure.Cause cause = validationFailure.getCauses().get(0);
+      Assert.assertEquals(CouchbaseConstants.NODES, cause.getAttribute(CauseAttributes.STAGE_CONFIG));
+      Assert.assertEquals(MOCK_STAGE_NAME, cause.getAttribute(STAGE));
+    }
+  }
+
+  @Test
+  public void testValidateBucketNull() {
```

Comment:
Could you please make validations in same way as for other plugins?
btw. none of your tests will fail in case if validation exception won't be thrown.

Reply:
All tests changed:
```
  @Test
   public void testValidateBucketNull() {
     CouchbaseSourceConfig config = CouchbaseSourceConfigBuilder.builder(VALID_CONFIG)
       .setBucket(null)
       .build();

     MockFailureCollector failureCollector = new MockFailureCollector(MOCK_STAGE);
     config.validate(failureCollector);
     assertValidationFailed(failureCollector, CouchbaseConstants.BUCKET);
   }
```